pyincore/dataset.py

```diff
@@ -343,8 +344,12 @@ def delete_temp_folder(self):
         """
         path = Path(self.local_file_path)
         absolute_path = path.parent.absolute()
+
         if os.path.isdir(absolute_path):
-            shutil.rmtree(absolute_path)
+            try:
+                shutil.rmtree(absolute_path)
+            except PermissionError as e:
```

Comment:
Since you are catch, would it be beneficial to catch more generic errors as well?

Reply:
I haven't found any other things to catch in here during my tests. Let me know if you want to add any exception, I will add. And, just adding general exception is not a very good idea